frontend/src/scenes/dashboard/Dashboard.tsx

```diff
@@ -59,6 +60,8 @@ function DashboardScene(): JSX.Element {
         useActions(dashboardLogic)
     const { groupsTaxonomicTypes } = useValues(groupsModel)
 
+    const [addInsightsToDashboardModalOpen, setAddInsightsToDashboardModalOpen] = useState<boolean>(false)
```

Comment:
Since we're using kea for state management, this should probably live in a kea logic as well. I think dashboardLogic is used here.

Then it might also not be necessary to props-pass the action down to the components, but instead use kea's `useAction` directly there (something like `const { addInsightsModalOpen } = useValues(dashboardLogic)`).